online-shop/src/app/components/add-product/add-product.component.ts

```diff
@@ -0,0 +1,71 @@
+import { HttpClient } from '@angular/common/http';
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { ActivatedRoute, Router } from '@angular/router';
+import { Product } from 'src/app/interfaces/productInterface';
+import { ErrorHandlerService } from 'src/app/services/error-handler.service';
+
+@Component({
+  selector: 'app-add-product',
+  templateUrl: './add-product.component.html',
+  styleUrls: ['./add-product.component.css'],
+})
+export class AddProductComponent implements OnInit {
+  constructor(
+    private route: ActivatedRoute,
+    private http: HttpClient,
+    private errorHandlerService: ErrorHandlerService,
+    private formBuilder: FormBuilder,
+    private router: Router
+  ) {}
+
+  ngOnInit(): void {}
+
+  product: Product = { id: -1, name: '', category: '', price: -1, description: '', image: '' };
```

Comment:
Product is used only in submit() , please make it a local variable. Also, using the constructor here to set the fields of the product is not necessary , since you overwrite them later. You can use var product = \<Product\>{}; instead.

Reply:
Done